src/components/forms/BookLoanForm.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from "react";
+
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Grid from "@mui/material/Grid";
+import Box from "@mui/material/Box";
+import Typography from "@mui/material/Typography";
+import Select from "@mui/material/Select";
+import MenuItem from "@mui/material/MenuItem";
+import FormControl from "@mui/material/FormControl";
+import InputLabel from "@mui/material/InputLabel";
+import { DesktopDatePicker } from "@mui/x-date-pickers/DesktopDatePicker";
+import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider";
+import { AdapterMoment } from "@mui/x-date-pickers/AdapterMoment";
+
+import axios from "axios";
+import { toast } from "react-toastify";
+import moment from "moment";
+
+const BookLoanForm = ({ bookLoan, handleClose }) => {
+  const [status, setStatus] = useState("");
+  const [dateBorrowed, setDateBorrowed] = useState(null);
+  const [dateDue, setDateDue] = useState(null);
+  const [dateReturned, setDateReturned] = useState(null);
+
+  useEffect(() => {
+    setStatus(bookLoan.status);
+    setDateBorrowed(bookLoan.date_borrowed);
+    setDateDue(bookLoan.date_due);
+    setDateReturned(bookLoan.date_returned);
+  }, [bookLoan]);
```

Comment:
ideally its not a good practice to store props in state. we should use props as it is and use a separately.